crime-means-assessment/src/test/java/uk/gov/justice/laa/crime/meansassessment/service/FullAssessmentAvailabilityServiceTest.java

```diff
@@ -0,0 +1,157 @@
+        assertThat(meansAssessmentResponse.getFullAssessmentAvailable()).isEqualTo(true);
+    }
+
+    @Test
+    public void whenResultIsFailAndCaseTypeIsIndictable_thenFullAssessmentAvailableIsTrue() {
+        meansAssessmentRequest.setAssessmentDate(null);
+        meansAssessmentRequest.setCaseType(CaseType.INDICTABLE);
+        meansAssessmentResponse.setResult("FAIL");
+
+        fullAssessmentAvailabilityService.processFullAssessmentAvailable(meansAssessmentRequest, meansAssessmentResponse);
+        assertThat(meansAssessmentResponse.getFullAssessmentAvailable()).isEqualTo(true);
+    }
+
+    @Test
+    public void whenResultIsFailAndCaseTypeIsCCAlready_thenFullAssessmentAvailableIsTrue() {
+        meansAssessmentRequest.setAssessmentDate(null);
+        meansAssessmentRequest.setCaseType(CaseType.CC_ALREADY);
+        meansAssessmentResponse.setResult("FAIL");
+
+        fullAssessmentAvailabilityService.processFullAssessmentAvailable(meansAssessmentRequest, meansAssessmentResponse);
+        assertThat(meansAssessmentResponse.getFullAssessmentAvailable()).isEqualTo(true);
+    }
+
+    @Test
+    public void whenResultIsFailAndCaseTypeIsAppealCC_thenFullAssessmentAvailableIsTrue() {
+        meansAssessmentRequest.setAssessmentDate(null);
+        meansAssessmentRequest.setCaseType(CaseType.APPEAL_CC);
+        meansAssessmentResponse.setResult("FAIL");
+
+        fullAssessmentAvailabilityService.processFullAssessmentAvailable(meansAssessmentRequest, meansAssessmentResponse);
+        assertThat(meansAssessmentResponse.getFullAssessmentAvailable()).isEqualTo(true);
+    }
+
+    @Test
+    public void whenResultIsFailAndCaseTypeIsEitherWayWithMagOutcomeNull_thenFullAssessmentAvailableIsTrue() {
+        meansAssessmentRequest.setAssessmentDate(null);
+        meansAssessmentRequest.setCaseType(CaseType.EITHER_WAY);
+        meansAssessmentResponse.setResult("FAIL");
+
+        fullAssessmentAvailabilityService.processFullAssessmentAvailable(meansAssessmentRequest, meansAssessmentResponse);
+        assertThat(meansAssessmentResponse.getFullAssessmentAvailable()).isEqualTo(false);
+    }
+
+    @Test
+    public void whenResultIsFailAndCaseTypeIsEitherWayWithMagOutcomeCommittedForTrial_thenFullAssessmentAvailableIsTrue() {
+        meansAssessmentRequest.setAssessmentDate(null);
+        meansAssessmentRequest.setCaseType(CaseType.EITHER_WAY);
+        meansAssessmentRequest.setMagCourtOutcome(MagCourtOutcome.COMMITTED_FOR_TRIAL);
+        meansAssessmentResponse.setResult("FAIL");
+
+        fullAssessmentAvailabilityService.processFullAssessmentAvailable(meansAssessmentRequest, meansAssessmentResponse);
+        assertThat(meansAssessmentResponse.getFullAssessmentAvailable()).isEqualTo(true);
+    }
+
+    @Test
+    public void whenResultIsFailAndCaseTypeIsEitherWayWithMagOutcomeAppealCC_thenFullAssessmentAvailableIsTrue() {
```

Comment:
The name needs to be changed to reflect the result.

Reply:
an oversight , resolved it.